c7n/resources/iam.py

```diff
@@ -695,6 +695,52 @@ def process(self, resources, event=None):
         return matched
 
 
+@User.filter_registry.register('group')
+class GroupPolicy(ValueFilter):
+    """Filter IAM users based on attached group values
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: iam-users-with-admin-access
+                resource: iam-user
+                filters:
+                  - type: group
+                    key: 'GroupName'
+                    value: 'AWSAdmin'
+    """
+
+    schema = type_schema('group', rinherit=ValueFilter.schema)
+    permissions = ('iam:ListGroupsForUser',)
+
+    def user_groups(self, user_set):
+        client = local_session(self.manager.session_factory).client('iam')
+        for u in user_set:
+            if 'c7n:Groups' not in u:
+                u['c7n:Groups'] = []
+            aps = client.list_groups_for_user(
+                UserName=u['UserName'])['Groups']
+            for ap in aps:
+                u['c7n:Groups'].append(
+                    client.get_group(GroupName=ap['GroupName'])['Group'])
```

Comment:
wondering if a) we can omit the group fetch, ie the most common match here is on name which is already present b) the groups are going to be common across users, so there's a lot of redundant fetching  on the groups.